community/kernel/src/main/java/org/neo4j/kernel/impl/newapi/Operations.java

```diff
@@ -149,10 +149,15 @@ public boolean nodeDelete( long node ) throws AutoIndexingKernelException
     }
 
     @Override
-    public long relationshipCreate( long sourceNode, int relationshipLabel, long targetNode )
+    public long relationshipCreate( long sourceNode, int relationshipLabel, long targetNode ) throws EntityNotFoundException
     {
         ktx.assertOpen();
-        throw new UnsupportedOperationException();
+        nodeExists( sourceNode );
```

Comment:
How is locking managed in this new implementation? Are the nodes already locked at this point?

Reply:
I would like it if in due time that was the case - that locking be handled through specific locking API calls at this level. However that is not the case right now, and I think it is a bit much to add such an API change before having completed the transition to the use of this API. Therefore I think we should make sure to handle locking in this implementation for now.

Reply:
...we do lock here for `nodeCreate` for example, so continued the pattern for now.